app/src/main/java/com/gathering/android/event/myevent/addevent/invitation/AddAttendeeBottomSheet.kt

```diff
@@ -0,0 +1,112 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(DialogFragment.STYLE_NORMAL, R.style.FullScreenCustomBottomSheet)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View {
+        binding = BottomSheetAddAttendeesBinding.inflate(LayoutInflater.from(requireContext()))
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.viewState.observe(viewLifecycleOwner) { state ->
+            when (state) {
+                AddAttendeesViewState.HideKeyboard -> hideKeyboard()
+                AddAttendeesViewState.CleaEditText -> binding.etAttendee.setText("")
+                is AddAttendeesViewState.SetAttendee -> binding.etAttendee.setText(state.attendee)
+                is AddAttendeesViewState.ShowError -> showToast(state.errorMessage ?: "")
+                is AddAttendeesViewState.AddAttendeeToRecyclerView -> adapter.addAttendeeItem(state.attendee)
+                is AddAttendeesViewState.RemoveAttendeeFromRecyclerView -> adapter.deleteAttendeeItem(
+                    state.attendee
+                )
+
+                is AddAttendeesViewState.NavigateToAddEvent -> {
+                    setNavigationResult(KEY_ARGUMENT_SELECTED_ATTENDEE_LIST, state.attendeeList)
+                    findNavController().popBackStack()
+                }
+
+                is AddAttendeesViewState.AddAttendeeButtonVisibility -> binding.btnAddAttendee.isEnabled =
+                    state.isAddAttendeeButtonEnabled
+            }
+        }
+
+        adapter.setOnAttendeeRemoveListener {
+            viewModel.onAttendeeRemoveItemClicked(it)
+        }
+
+        binding.rvContact.adapter = adapter
+        binding.rvContact.addItemDecoration(
+            DividerItemDecoration(
+                context, DividerItemDecoration.HORIZONTAL
+            )
+        )
+
+        binding.etAttendee.doOnTextChanged { text, _, _, _ ->
+            viewModel.onAttendeeEmailChanged(text.toString())
+        }
+
+        binding.btnOk.setOnClickListener {
+            val attendeeList = adapter.getAttendeeItems()
+            viewModel.onOKButtonClicked(attendeeList)
+        }
+
+        binding.btnAddAttendee.setOnClickListener {
+            viewModel.onAddAttendeeButtonClicked(binding.etAttendee.text.toString())
+        }
+
+
+        val attendees = arguments?.getString("attendee_list")
+        viewModel.onViewCreated(attendees)
+
+    }
+
+    private fun hideKeyboard() {
+        val inputSystemService = requireContext().getSystemService(Context.INPUT_METHOD_SERVICE)
+        val imm = inputSystemService as? InputMethodManager
+        imm?.hideSoftInputFromWindow(view?.windowToken, 0)
+    }
+
+    private fun showToast(text: String) {
```

Comment:
can we use the extension function you created for showing toast called "showErrorText" here as well? instead of this function:
private fun showToast(text: String) {
  Toast.makeText(
            requireContext(), text, Toast.LENGTH_SHORT
        ).show()
}

Reply:
Ya, I think so. we can use that extension instead, There are more show toasts in our project that should be removed.
Thank you for mentioning that :)

Reply:
you are absolutely right. and thank you for creating that extension function because I also used it in my previous ticket.